internal/staging/collect_result.go

```diff
@@ -453,3 +481,200 @@ func (c *controller) getReverificationStatus(queue *s2hv1beta1.Queue) rpc.Compon
+
+		pods := &corev1.PodList{}
+		if err := c.client.List(context.TODO(), pods, listOpt); err != nil {
+			logger.Error(err, "cannot list pods")
+			return err
+		}
+
+		jobs := &batchv1.JobList{}
+		if err := c.client.List(context.TODO(), jobs, listOpt); err != nil {
+			logger.Error(err, "cannot list jobs")
+			return err
+		}
+
+		c.extractDeploymentIssues(pods, jobs, deploymentIssuesMaps)
+	}
+
+	deploymentIssues := c.convertToDeploymentIssues(deploymentIssuesMaps)
+	queue.Status.SetDeploymentIssues(deploymentIssues)
+
+	return nil
+}
+
+func (c *controller) extractDeploymentIssues(pods *corev1.PodList, jobs *batchv1.JobList,
+	issuesMaps map[s2hv1beta1.DeploymentIssueType][]s2hv1beta1.FailureComponent) {
+
+	for _, pod := range pods.Items {
+		compName := c.extractComponentNameFromPod(pod)
+		failureComp := s2hv1beta1.FailureComponent{
+			ComponentName: compName,
+		}
+
+		// ignore job pod
+		jobFound := false
+		for _, podRef := range pod.OwnerReferences {
+			if strings.ToLower(podRef.Kind) == "job" {
+				jobFound = true
+				break
+			}
+		}
+
+		if !jobFound {
+			// check init container issue
+			initContainerStatuses := pod.Status.InitContainerStatuses
+			initFound := false
+			for _, initContainerStatus := range initContainerStatuses {
+				if !initContainerStatus.Ready {
+					initFound = true
+					failureComp.FirstFailureContainerName = initContainerStatus.Name
+					failureComp.RestartCount = initContainerStatus.RestartCount
+					c.appendDeploymentIssues(s2hv1beta1.DeploymentIssueWaitForInitContainer, failureComp, issuesMaps)
+				}
+			}
+			if initFound {
+				continue
+			}
+
+			containerStatuses := pod.Status.ContainerStatuses
+			found := false
+			for _, containerStatus := range containerStatuses {
+				if !containerStatus.Ready {
+					failureComp.FirstFailureContainerName = containerStatus.Name
+					failureComp.RestartCount = containerStatus.RestartCount
+
+					waitingState := containerStatus.State.Waiting
+					if waitingState != nil {
+						switch waitingState.Reason {
+						// check ImagePullBackOff issue
+						case "ImagePullBackOff", "ErrImagePull":
+							c.appendDeploymentIssues(s2hv1beta1.DeploymentIssueImagePullBackOff, failureComp, issuesMaps)
+							found = true
+
+						case "CrashLoopBackOff", "Error":
+							c.appendDeploymentIssues(s2hv1beta1.DeploymentIssueCrashLoopBackOff, failureComp, issuesMaps)
+							found = true
+
+						case "ContainerCreating":
+							c.appendDeploymentIssues(s2hv1beta1.DeploymentIssueContainerCreating, failureComp, issuesMaps)
+							found = true
+						default:
+							c.appendDeploymentIssues(s2hv1beta1.DeploymentIssueUndefined, failureComp, issuesMaps)
+							found = true
+						}
+					}
+
+					if found {
+						break
+					}
+
+					runningState := containerStatus.State.Running
+					if runningState != nil {
+						// if running 0/1, count as CrashLoopBackOff
+						if containerStatus.RestartCount > 0 {
```

Comment:
If component is in running 0/1. and restartcount == 0. should we handle it?

Reply:
it will be treated as undefined type, since means that readiness failed but liveness is good

Reply:
and it might mean that it cannot be started in time without any issues

Reply:
@quraynai oh, I haven't handled it yet, will update